src/app/operators/dialogs/create-rgb/create-rgb.component.ts

```diff
@@ -0,0 +1,135 @@
+    styleUrls: ['./create-rgb.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CreateRgbComponent implements OnInit, OnDestroy {
+    readonly inputTypes = [ResultTypes.RASTER];
+    readonly numberOfRasters = 3;
+
+    form: FormGroup;
+
+    private inputLayersubscriptions: Subscription;
+
+    constructor(private projectService: ProjectService,
+                private notificationService: NotificationService) {
+    }
+
+    ngOnInit() {
+        this.form = new FormGroup({
+            'inputLayers': new FormControl(
+                undefined,
+                [Validators.required, Validators.minLength(this.numberOfRasters), Validators.maxLength(this.numberOfRasters)],
+            ),
+            'redMin': new FormControl(undefined, [Validators.required]),
+            'redMax': new FormControl(undefined, [Validators.required]),
+            'redScale': new FormControl(1, [Validators.required, Validators.min(0), Validators.max(1)]),
+            'greenMin': new FormControl(undefined, [Validators.required]),
+            'greenMax': new FormControl(undefined, [Validators.required]),
+            'greenScale': new FormControl(1, [Validators.required, Validators.min(0), Validators.max(1)]),
+            'blueMin': new FormControl(undefined, [Validators.required]),
+            'blueMax': new FormControl(undefined, [Validators.required]),
+            'blueScale': new FormControl(1, [Validators.required, Validators.min(0), Validators.max(1)]),
+        });
+
+        this.inputLayersubscriptions = this.form.controls['inputLayers'].valueChanges
+            .subscribe((inputLayers: Array<RasterLayer<RasterSymbology>>) => { // set meaningful default values if possible
+                const colors = ['red', 'green', 'blue'];
+                inputLayers.forEach((inputRaster, i) => {
+                    if (inputRaster && !this.form.controls[`${colors[i]}Min`].value) {
+                        this.form.controls[`${colors[i]}Min`].setValue(inputLayers[i].operator.getDataType('value').getMin());
+                    }
+                    if (inputRaster && !this.form.controls[`${colors[i]}Max`].value) {
+                        this.form.controls[`${colors[i]}Max`].setValue(inputLayers[i].operator.getDataType('value').getMax());
+                    }
+                });
+            });
+    }
+
+    ngOnDestroy() {
+        if (this.inputLayersubscriptions) {
+            this.inputLayersubscriptions.unsubscribe();
+        }
+    }
+
+    add() {
+        const inputs: Array<RasterLayer<RasterSymbology>> = this.form.controls['inputLayers'].value;
+        const operators = inputs.map(layer => layer.operator);
+
+        if (inputs.length !== 3) {
+            this.notificationService.error('RGBA calculation requires 3 inputs.');
+            return;
+        }
+        if (unequalProjections(operators)) {
+            this.notificationService.error('Input rasters must be of same projection.');
```

Comment:
this deviates from the design choice to add projections into the workflow on the fly. However, i am fine with this for the desired featureset.

Reply:
see https://github.com/umr-dbs/wave/pull/12/commits/fe7615bae66897ef93a65d1b2d263b1a30ce59cb